vcstool/commands/compare.py

```diff
@@ -1,80 +1,595 @@
 from vcstool.streams import set_streams
+from vcstool.outputs import CompareOutput
 
 from .command import add_common_arguments
 from .command import Command
 
 
 class CompareCommand(Command):
 
-    command = 'compare'
-    help = 'Compare working copy to the repository list file'
+    command = "compare"
+    help = "Compare working copy to the repository list file"
 
-    def __init__(self, args):
-        super(CompareCommand, self).__init__(args)
+    def __init__(self, args: argparse.Namespace) -> None:
+        super().__init__(args)
+        self.progress = args.progress
+        self.workers = args.workers
+        self.debug = args.debug
 
+    @classmethod
+    def get_parser(cls) -> argparse.ArgumentParser:
+        parser = argparse.ArgumentParser(description=f"{cls.help}", prog="vcs compare")
+        group = parser.add_argument_group('"compare" command parameters')
+        group.add_argument(
+            "-i",
+            "--input",
+            type=str,
+            default="workspace.repos",
+            help="Path to the repository list file for the workspace.",
+        )
+        group.add_argument(
+            "-p",
+            "--progress",
+            action="store_true",
+            default=False,
+            help="Show progress of the jobs during execution.",
+        )
+        group.add_argument(
+            "-n",
+            "--nested",
+            action="store_true",
+            default=False,
+            help="Search the workspace for nested repositories.",
+        )
+        group.add_argument(
+            "-s",
+            "--significant",
+            action="store_true",
+            default=False,
+            help="Only show significant repos.",
+        )
+        return parser
 
-def get_parser():
-    parser = argparse.ArgumentParser(
-        description='Compare working copy to the repository list file',
-        prog='vcs compare')
-    group = parser.add_argument_group('"compare" command parameters')
-    group.add_argument(
-        '--input', type=argparse.FileType('r'), default='-')
-    return parser
+    def execute(self) -> List[Dict[str, Any]]:
+        """Convenience method which executes this CompareCommand using a number of workers."""
+        clients = find_repositories(self.paths, nested=self.nested)
+        jobs = generate_jobs(clients, self)
+        return execute_jobs(
+            jobs,
+            show_progress=self.progress,
+            number_of_workers=self.workers,
+            debug_jobs=self.debug,
+        )
 
 
-def main(args=None, stdout=None, stderr=None):
-    set_streams(stdout=stdout, stderr=stderr)
+class Legend:
+    """Namespace containing symbols used in the CompareTable."""
 
-    parser = get_parser()
-    add_common_arguments(parser, skip_hide_empty=True,
-                         skip_nested=True, skip_repos=True, path_nargs='?')
-    args = parser.parse_args(args)
+    # fmt: off
+    SUBMODULE        = "c"
```

Comment:
Super duper nit: Maybe better to call this `REPO` and `"r"`

Reply:
Agree I definitely intended to do something like this, just forgot about it. Thanks!